tools/build_cros_gcc

```diff
@@ -1,178 +1,211 @@
- ./$MPFR \
- ./$MPC \
- ./linux-$KERNEL_VERSION \
- ./$ISL \
- ./$GLIBC.tar.xz \
- ./$GMP.tar.xz \
- ./$GCC.tar.xz \
- ./$BINUTILS.tar.xz \
- ./linux-$KERNEL_VERSION.tar.xz \
- ./$MPFR.tar.xz \
- ./$CLOOG.tar.gz \
- ./$ISL.tar.bz2 \
- ./$MPC.tar.gz || true
- shift
- exit 0;;
- -t|--const)
- echo KERNEL_VERSION=$KERNEL_VERSION
- echo MAJOR_VERSION=$MAJOR_VERSION
- echo CLOOG=$CLOOG
- echo ISL=$ISL
- echo MPC=$MPC
- echo GMP=$GMP
- echo MPFR=$MPFR
- echo GCC=$GCC
- echo BINUTILS=$BINUTILS
- echo GLIBC=$GLIBC
- shift
- exit 0;;
- armv7l-cros-linux-gnueabihf)
- true;;
- aarch64-cros-linux-gnu)
- true;;
- i686-cros-linux-gnu)
- true;;
- x86_64-cros-linux-gnu)
- true;;
- *)
- echo Invalid target or option \'$1\'
- $0 -h
- exit 1;;
+ case x${1}x in
+   x-hx|x--helpx) echo Usage:
+     echo \ $0 target [jobs] [prefix]
+     echo \ \-h, --help: Display this help
+     echo \ \-v, --version: Display version
+     echo \ \-t, --const: Display constants
+     echo \ \-c, --clean: Clean current directory
+     echo \ Valid targets: 'aarch64-cros-linux-gnu', 'x86_64-cros-linux-gnu', 'i686-cros-linux-gnu', 'armv7l-cros-linux-gnueabihf'
+     exit 0;;
+   x-vx|x--versionx) echo build_cros_gcc version 1.27.1
+     exit 0;;
+   x-cx|x--cleanx) rm -rf build-binutils build-gcc build-glibc
+     rm -rf \
+     ./${GCC} \
+     ./${CLOOG} \
+     ./${GMP} \
+     ./${GLIBC} \
+     ./${BINUTILS} \
+     ./${MPFR} \
+     ./${MPC} \
+     ./linux-${KERNEL_VERSION} \
+     ./${ISL} \
+     ./${GLIBC}.tar.xz \
+     ./${GMP}.tar.xz \
+     ./${GCC}.tar.xz \
+     ./${BINUTILS}.tar.xz \
+     ./linux-${KERNEL_VERSION}.tar.xz \
+     ./${MPFR}.tar.xz \
+     ./${CLOOG}.tar.gz \
+     ./${ISL}.tar.bz2 \
+     ./${MPC}.tar.gz \
+     ./sha256sum || true
+     exit 0;;
+   x-tx|x--constx)
+     echo KERNEL_VERSION=${KERNEL_VERSION}
+     echo MAJOR_VERSION=${MAJOR_VERSION}
+     echo CLOOG=${CLOOG}
+     echo ISL=${ISL}
+     echo MPC=${MPC}
+     echo GMP=${GMP}
+     echo MPFR=${MPFR}
+     echo GCC=${GCC}
+     echo BINUTILS=${BINUTILS}
+     echo GLIBC=${GLIBC}
+     exit 0;;
+   xarmv7l-cros-linux-gnueabihfx)
+     true;;
+   xaarch64-cros-linux-gnux)
+     true;;
+   xi686-cros-linux-gnux)
+     true;;
+   xx86_64-cros-linux-gnux)
+     true;;
+   xx)
+     $0 -h
+     exit 1;;
+   x*x)
+     echo -e "\033[1;31mInvalid target or option '$1'\033[0m"
+     $0 -h
+   exit 1;;
  esac
 
- # '\033[1;31' outputs red text. '\033[0m' reverts it to normal.
- command -v crew > /dev/null || { echo "\033[1;31mcrew command not found.\033[0m"; exit 1; }
+ TARGET="$1" && shift
 
- if test ! $2; then
-   echo "$1" | grep x86_64 1>/dev/null
-   if [ $? == 0 ]; then
-   $0 $1 x86 $3;
-   exit 0;
+ if ! [[ "$1" =~ "^[0-9]+$" ]] &>/dev/null; then
+   JOBS="$1"
+   shift
+ else
+   shift
+ fi
+ if [[ -d $1 ]] &>/dev/null; then
+   PREFIX="$1"
+   shift
+ else
+   if ! [[ "$1" =~ "^[0-9]+$" ]] &>/dev/null; then
+     JOBS="$1"
+     shift
+   else if { mkdir -p "$1" &>/dev/null; }; then
```

Comment:
This is not recommended.  Why do you need to create a directory?

Reply:
I don't. I'll fix that.

Reply:
Fixed.